### PR TITLE
Fix #20

### LINKED ISSUE #20: Tests faild on Windows host
## Issue summary

While running `composer test` on Windows OS, some test do not pass (1 failed, 2 errors).
php-cs-fixer starts run when I change phpcs command in composer.json from:
```
php-cs-fixer fix -v --diff --dry-run --allow-risky=yes;
```
to:
```
php-cs-fixer fix -v --diff --dry-run --allow-risky=yes
```
(no semicolon on end)

### System informations

| Information | Description |
|--------------|---------|
| league-uri-parser version | dev-master  |
| PHP/HHVM version | PHP 7.3.6 with Xdebug 2.7.2 (ext-intl installed) |
| OS Platform | Windows 10 |

### Expected result
Something like this:
```
> php-cs-fixer fix -v --diff --dry-run --allow-risky=yes;
Loaded config default from "/path/uri-parser/.php_cs".
Using cache file ".php_cs.cache".
SSSSSS
Legend: ?-unknown, I-invalid file syntax, file ignored, S-Skipped, .-no changes, F-fixed, E-error

Checked all files in 0.003 seconds, 10.000 MB memory used
> phpstan analyse -l max -c phpstan.src.neon src
 3/3 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓] 100%

 3/3 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓] 100% [OK] No errors

> phpstan analyse -l max -c phpstan.tests.neon tests
 1/1 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓] 100%

 [OK] No errors

> phpunit --coverage-text
PHPUnit 8.3.4 by Sebastian Bergmann and contributors.

Runtime:       PHP 7.3.8
Configuration: /path/uri-parser/phpunit.xml
Error:         No code coverage driver is available

...............................................................  63 / 124 ( 50%)
.............................................................   124 / 124 (100%)

Time: 97 ms, Memory: 4.00 MB

OK (124 tests, 124 assertions)
```
### Actual result
```
> php-cs-fixer fix -v --diff --dry-run --allow-risky=yes
Loaded config default from "C:\path\uri-parser\.php_cs".
Using cache file ".php_cs.cache".
SSSS.S
Legend: ?-unknown, I-invalid file syntax, file ignored, S-Skipped, .-no changes, F-fixed, E-error

Checked all files in 0.113 seconds, 12.000 MB memory used
> phpstan analyse -l max -c phpstan.src.neon src
 3/3 [============================] 100%

 [OK] No errors

> phpstan analyse -l max -c phpstan.tests.neon tests
 1/1 [============================] 100%

 [OK] No errors

> phpunit --coverage-text
PHPUnit 8.3.4 by Sebastian Bergmann and contributors.

Runtime:       PHP 7.3.6 with Xdebug 2.7.2
Configuration: C:\Projects\uri-parser\phpunit.xml

...............................................E...............  63 / 124 ( 50%)
...........F.............................................E...   124 / 124 (100%)

Time: 1.49 seconds, Memory: 8.00 MB

There were 2 errors:

1) LeagueTest\Uri\UriStringTest::testParseSucced with data set "RFC3987 example" ('http://bébé.bé./有词法别名.zh', array('http', null, null, 'bébé.bé.', null, '/有词法别名.zh', null, null))
League\Uri\Exception\MalformedUri: Host `bébé.bé.` is not a valid IDN host : a label or domain name contains disallowed characters.

C:\Projects\uri-parser\src\UriString.php:448
C:\Projects\uri-parser\src\UriString.php:382
C:\Projects\uri-parser\src\UriString.php:342
C:\Projects\uri-parser\src\UriString.php:309
C:\Projects\uri-parser\tests\UriStringTest.php:34

2) LeagueTest\Uri\UriStringTest::testBuild with data set "RFC3987 example" ('http://bébé.bé./有词法别名.zh', 'http://bébé.bé./有词法别名.zh')
League\Uri\Exception\MalformedUri: Host `bébé.bé.` is not a valid IDN host : a label or domain name contains disallowed characters.

C:\Projects\uri-parser\src\UriString.php:448
C:\Projects\uri-parser\src\UriString.php:382
C:\Projects\uri-parser\src\UriString.php:342
C:\Projects\uri-parser\src\UriString.php:309
C:\Projects\uri-parser\tests\UriStringTest.php:792

--

There was 1 failure:

1) LeagueTest\Uri\UriStringTest::testParseFailed with data set "invalid Host with fullwith (1)" ('http://％００.com')
Failed asserting that exception of type "League\Uri\Exception\MalformedUri" is thrown.

ERRORS!
Tests: 124, Assertions: 122, Errors: 2, Failures: 1.

Generating code coverage report in Clover XML format ... done [73 ms]

Generating code coverage report in HTML format ... Script phpunit --coverage-text handling the phpunit event returned with error code 2
Script @phpunit was called via test
done [223 ms]


Code Coverage Report:
  2019-09-11 15:07:53

 Summary:
  Classes: 100.00% (1/1)
  Methods: 100.00% (8/8)
  Lines:   100.00% (110/110)

\League\Uri::League\Uri\UriString
  Methods: 100.00% ( 8/ 8)   Lines: 100.00% (110/110)
```

### PR DESCRIPTION
**Pull request by pawelkg**
## Introduction

Tests Faild on Windows host due to invalid results of strtolower function.

### Describe the new/upated/fixed feature

This PR fix issue #20 

### Backward Incompatible Changes

None

### Targeted release version

1.4.2

### PR Impact

None

## Open issues

None

### CODE REVIEW

File: src/UriString.php [hunk 1]
```diff
@@ -406,7 +406,7 @@ private static function filterRegisteredName(string $host): string
         $idn_support = $idn_support ?? function_exists('idn_to_ascii') && defined('INTL_IDNA_VARIANT_UTS46');
         // @codeCoverageIgnoreEnd
 
-        $formatted_host = rawurldecode(strtolower($host));
```
Comment (nyamsprod) [inline src/UriString.php:409]: why was this change added ? What is the relation with the php-cs-fixer issue ?

Comment (pawelkg) [reply src/UriString.php:409]: To run tests on Windows You need at first place fix command in composer.json. If there is semicolon at end of command this error is show:
```
[PhpCsFixer\ConfigurationException\InvalidConfigurationException (16)]
  The rules contain risky fixers (psr0, psr4), but they are not allowed to run. Perhaps you forget to use --allow-risky=yes option?
```
Code do not recognize options correctly.
When tests runs, they output errors describer in PR on Windows. Those errors are because strtolower function behave different on Windows and Linux. The example code:
```php
$test = '％００.com';
echo $test;
echo strtolower($test);
echo rawurldecode(strtolower($test));
```
produce on Linux:
```
％００.com
％００.com
％００.com
```
and on Windows:
```
％００.com
ﾅﾐﾐ.com
ﾅﾐﾐ.com
```

While on Linux it's still invalid domain name, on Windows it's can be converted to punny-code.

**Comment by nyamsprod** [comment]
OK understood. Now I have a good and a bad news for you :'( 

The bad news is that your PR is against the wrong branch. The master branch has been merged in the https://github.com/thephpleague/uri which I'm about to release. The master branch contains many BC breaks vs the 1.x series so your patch as it stand can never hit the 1.x series 😢 . I'm sorry about that.

The 1.4.3 which is the one you want to target should be in the 1.x series. The good news is I can create such branch and make it the default so that you can apply your changes there and even release  a 1.4.3 fix, I'm not against it.

The reason for the move was maintenance and consistency. The project had too many decoupled component and a fix in just one had too many effect on the other components so I have regrouped the uri parser with the uri package for better coherence.

If you are ok with that you need to give me a couple of day so I can prepare correctly the changes in the repository for your PR to be accepted.

**Comment by pawelkg** [comment]
OK. So I will move in my project to uri repo. I think it's good to implement this fix to this repo, meaby someone will take usage of it.
BTW. I prepare some other changes to this repo avaliable at my branch: https://github.com/pawelkg/uri-parser/tree/fixes-and-updates
Should I create PR for this changes also?

**Comment by nyamsprod** [comment]
@pawelkg If you think these changes are valuable to the library of course. They will be reviewed and merged if they are found useful 😉

**Comment by nyamsprod** [comment]
a fix for the composer script as landed on the 1.x branch which will be used only for maintenace of the 1.x series. For any new feature please refer to the League/Uri package directly.
